src/Traits/Installer.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Native\Electron\Traits;
+
+trait Installer
+{
+    protected function installNPMDependencies()
+    {
+        if ($this->option('force') || $this->confirm('Would you like to install the NativePHP NPM dependencies?', true)) {
+            $this->comment('Installing NPM dependencies (This may take a while)...');
+
+            if (!this->option('installer')) {
+                $this->info('Installing NPM dependencies using the npm package manager...');
+                $this->installWithNPM();
+            }
+
+            switch ($this->option('installer')) {
+                case 'yarn':
+                    $this->info('Installing NPM dependencies using the yarn package manager...');
+                    $this->installWithYarn();
+                    break;
+                case 'pnpm':
+                    $this->info('Installing NPM dependencies using the pnpm package manager...');
+                    $this->installWithPNPM();
+                    break;
+                default:
+                    $this->info('Installing NPM dependencies using the npm package manager...');
+                    $this->installWithNPM();
+            }
```

Comment:
maybe you could change this to the following:
```php
switch ($installer = $this->option('installer')) {
    case 'npm':
        $this->info('Installing NPM dependencies using the npm package manager...');
        $this->installWithNPM();
        break;
    default:
        $this->info("Installing NPM dependencies using the {$installer} package manager...");
        $this->installWithCustom($installer);
}

protected function installWithCustom(?string $installer = null)
{
    $installer ??= $this->option('installer');

    $this->executeCommand("{$installer} install", $this->nativePhpPath());
}
```
This way a user could use custom package managers without having to add code.

You could also make an array with installers and the preffered command, for example:
```php
[
    'npm' => 'npm set progress=false && npm install',
    ...
]
```
This way you could add any command when u want to set a specific install command for a certain package manager